azkaban-common/src/main/java/azkaban/scheduler/TriggerBasedScheduleLoader.java

```diff
@@ -97,18 +98,14 @@ private Condition createTriggerCondition(Schedule s) {
     return cond;
   }
 
-  // if failed to trigger, auto expire?
   private Condition createExpireCondition(Schedule s) {
-    Map<String, ConditionChecker> checkers =
-        new HashMap<String, ConditionChecker>();
-    ConditionChecker checker =
-        new BasicTimeChecker("BasicTimeChecker_2", s.getFirstSchedTime(),
-            s.getTimezone(), s.isRecurring(), s.skipPastOccurrences(),
-            s.getPeriod(), s.getCronExpression());
+    Map<String, ConditionChecker> checkers = new HashMap<>();
+    ConditionChecker checker = new BasicTimeChecker("EndTimeChecker_1", s.getFirstSchedTime(),
```

Comment:
Thanks for your review. Do you have any good idea what name could be better here? @chengren311